src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Build.ps1

```diff
@@ -1958,62 +1958,82 @@ class Build: ADOSVTBase
         return $controlResult
     }
 
-    hidden [ControlResult] CheckForkedBuildTrigger([ControlResult] $controlResult)
-    {
+    hidden [ControlResult] CheckForkedBuildTrigger([ControlResult] $controlResult) {
         $controlResult.VerificationResult = [VerificationResult]::Failed
+        $projectVisibilityCheck = @($this.ControlSettings.Project.ProjectVisibilitiesToCheck)
 
-        if([Helpers]::CheckMember($this.BuildObj[0],"triggers"))
-        {
-            $pullRequestTrigger = $this.BuildObj[0].triggers | Where-Object {$_.triggerType -eq "pullRequest"}
-
-            # initlizing $isRepoPrivate = $true as visibility setting is not available for ADO repositories.
-            $isRepoPrivate = $true
-            if ([Helpers]::CheckMember($this.BuildObj[0],"repository.properties.IsPrivate")) {
-                $isRepoPrivate = $this.BuildObj[0].repository.properties.IsPrivate
-            }
+        if ($projectVisibilityCheck.Count -eq 0) {
+            $controlResult.AddMessage([VerificationResult]::Error, "Project visibilities to check are not available in control settings");
+            return $controlResult
+        }
 
-            if($pullRequestTrigger)
-            {
-                if(($isRepoPrivate -eq $false) -and [Helpers]::CheckMember($pullRequestTrigger,"forks"))
-                {
+        if ($projectVisibilityCheck -notcontains $this.ResourceContext.ResourceDetails.project.visibility) {
+            $controlResult.AddMessage([VerificationResult]:: Verify), "Project is of visibility: " + $this.ResourceContext.ResourceDetails.project.visibility + ". Current control scan is only applicable for private projects."
+            return $controlResult
+        }
+        
+        if ([Helpers]::CheckMember($this.BuildObj[0], "triggers")) {
+            $pullRequestTrigger = $this.BuildObj[0].triggers | Where-Object { $_.triggerType -eq "pullRequest" }
 
-                    if(($pullRequestTrigger.forks.enabled -eq $true) -and ($pullRequestTrigger.forks.allowSecrets -eq $true))
-                    {
-                        $controlResult.AddMessage([VerificationResult]::Failed,"Pipeline secrets are marked as available to pull request validations of public repo forks.");
-                        if ($this.ControlFixBackupRequired -or $this.BaselineConfigurationRequired)
-                        {
-                            #Data object that will be required to fix the control
-                            $controlResult.BackupControlState = $pullRequestTrigger.forks.allowSecrets;
+            if ($pullRequestTrigger) {
+                if ($pullRequestTrigger.pipelineTriggerSettings.forkProtectionEnabled -eq $true) {
+                    if (!$pullRequestTrigger.pipelineTriggerSettings.buildsEnabledForForks) {
```

Comment:
Please try to add proper comments wherever needed. It would be helpful for everyone to understand.